src/Arr.php

```diff
@@ -162,9 +160,9 @@ public function prev()
      *
      * @link https://php.net/manual/en/iterator.key.php
      *
-     * @return string|float|int|bool|null scalar on success, or null on failure.
+     * @return string|int|null scalar on success, or null on failure.
      */
-    public function key(): mixed
+    public function key(): string|int|null
```

Comment:
Hi @Tautve , are you sure you want to "limit" the key type to string, int or null?

Reply:
well, I would say so, because they 'key()' func, used below, from standard_8.php looks like this:'function key(object|array $array): string|int|null {}', so, its return types are the same. I mean, your func 'key' wouldn't return anything else except the same types declared in the original function

Reply:
ok, @Tautve it makes sense. thank you for the explanation.
going to accept it, I see that the tests fail.
Did you launch tests before opening the PR?

Reply:
Sorry, my bad, missed to add `array` as return type to Arr::keys()

Reply:
thank you !!!! Great Pull Request!